platform/measurement_utils.cpp

```diff
@@ -23,12 +23,31 @@ using namespace settings;
 using namespace std;
 using namespace strings;
 
+// Global Locale pointer to store system locale.
+Locale *pSysLocale = NULL;
```

Comment:
The pointer is not needed.
```suggestion
static Locale g_systemLocale;
```

Reply:
Can it be also updated from iOS?

Reply:
> Can it be also updated from iOS?

@kirylkaveryn, @fabwu: PTAL at the implementation to detect locale change for iOS. I've tried to make some tests with a simulator, but everytime that I change de iOS language/locate, the app is restarted so the `currentLocaleDidChange` function is never called...

Reply:
@gpesquero right, I forgot that on iOS all apps are restarted when the locale is changed, so it is not needed.

Are there any alternative solutions to fix unit tests? These locales are complicating the code a lot...

Reply:
> Are there any alternative solutions to fix unit tests? These locales are complicating the code a lot...

Doing all the locale formatting in the Android/iOS part will make all the c++ code and related unit tests much cleaner, but we may encounter performance issues if we do this _quite often_ distance & speed localized formatting using non-native functions (remember the checks that I did in https://github.com/organicmaps/organicmaps/pull/4029#issuecomment-1646971596).

Let me make a few tests & measurements with a simple formatting in Android and making a performance comparison vs c++ implementation. I will get back to you once I made the tests.

We shall also review the update rate of the distance & speed while routing. Right now they are updated 20 times per second; IMO, this update rate shall be lower (5 times per second?)

Reply:
@gpesquero If I understand this correctly you need to change the locale for unit tests, so why don't you add a default parameter to `ToStringPrecision()` and overwrite the locale in the test?

Reply:
> @gpesquero If I understand this correctly you need to change the locale for unit tests, so why don't you add a default parameter to `ToStringPrecision()` and overwrite the locale in the test?

Hi @fabwu: We already have 2 possible ways to call `ToStringPrecision()`: one with locale as an argument and the other uses an internal variable to store the system local that it's only updated once. This PR just adds a global variable to store the system locale, and that would allow us to change the locale during unit tests for Distance/Speed calls from c++, Android & iOS.

Reply:
> @gpesquero right, I forgot that on iOS all apps are restarted when the locale is changed, so it is not needed.

I've removed the locale change detection for iOS.